src/WatchDogs.Infrastructure.FakeSource/FakeSourceWatcher.cs

```diff
@@ -60,9 +65,35 @@ private async Task LoadFakeDataEverySecondAsync()
         {
             while (await _timer.WaitForNextTickAsync(_cts.Token))
             {
+                var unitOfWorkFactory = new EntityFrameworkUnitOfWorkFactory(_dbContextOptions);
```

Comment:
This is quite inefficient. You are creating a factory after each tick but nothing is changing in the context of that. Why not inject it via DI instead?

Reply:
It should be working by now. I had quite some trouble with that because I didn't know how this factory pattern work, now it's much clearer.